packages/@lwc/features/src/babel-plugin/index.js

```diff
@@ -65,47 +73,51 @@ module.exports = function({ types: t }) {
-                    if (
-                        objectPath.isIdentifier({ name: this.defaultSpecifierName }) &&
-                        propertyPath.isIdentifier()
-                    ) {
-                        const binding = this.importDeclarationScope.getBinding(
-                            objectPath.node.name
-                        );
-                        // If this thing is an actual reference to the imported feature flag lookup.
-                        if (binding && binding.referencePaths.includes(objectPath)) {
-                            const name = propertyPath.node.name;
-                            const value = this.featureFlags[name];
-                            validate(name, value);
-                            if (!this.opts.prod || value === null) {
-                                testPath.replaceWithSourceString(
-                                    `${RUNTIME_FLAGS_IDENTIFIER}.${name}`
-                                );
-                            } else if (value === true) {
-                                // Transform the IfStatement into a BlockStatement
-                                path.replaceWith(path.node.consequent);
-                            } else if (value === false) {
-                                // Remove IfStatement
-                                path.remove();
-                            }
-                        }
-                    }
+                const isUnaryNegation = testPath.isUnaryExpression({ operator: '!' });
+                if (isUnaryNegation) {
+                    testPath = testPath.get('argument');
                 }
 
-                // Transform runtime flags into compile-time flags, where appropriate, for
-                // production mode. This serves to undo the non-production mode transform of
-                // forcing all flags to be runtime flags.
-                if (this.opts.prod && isRuntimeFlag(testPath, this.featureFlags)) {
-                    const name = testPath.node.property.name;
-                    const value = this.featureFlags[name];
-                    validate(name, value);
+                if (!testPath.isMemberExpression()) {
+                    return;
+                }
+
+                const objectPath = testPath.get('object');
+                const propertyPath = testPath.get('property');
+                const isRuntimeFlag = objectPath.isIdentifier({ name: RUNTIME_FLAGS_IDENTIFIER });
+
+                let isCompileTimeFlag = false;
+                if (this.defaultImportPath) {
+                    isCompileTimeFlag = objectPath.isIdentifier({
+                        name: this.defaultImportPath.node.local.name,
+                    });
+                }
+
+                // If the member expression is not a feature flag lookup
+                if (!isRuntimeFlag && !isCompileTimeFlag) {
```

Comment:
I'm a bit confused how the runtime flag plays into this? I maybe mixing concept by thinking about the fact that we disallow importing and using features flags manually

Reply:
This is just an optimization to ignore the visited node if the object of the member expression is neither `featureFlags` nor `runtimeFlags` because we're running for any member expression inside an if-statement test (e.g., `if (foo.bar) { ... }`).

I'll add more detail to the comment to try and address your confusion.